### ISSUE TITLE
grid bug with multiple selectable values only using ajax

### ISSUE THREAD

**Issue by niphlod** [web2py/web2py#750]
_From [manuele...._at_gmail.com](https://code.google.com/u/103187076317995702211/) on January 13, 2015 16:04:37_

\## What steps will reproduce the problem?

1. Consider the subsequent code:

* In a model source (i.e. models/test.py)

```python
\# -*- coding: utf-8 -*-
from gluon.contrib.populate import populate

db.define_table('generic',
    Field('first_name')
)

if db(db.generic.id>0).count()==0:
    populate(db.generic,100)

def _foo(ids):
    print ids
````

* In a controller file (i.e. controllers/test.py)

```python
\# -*- coding: utf-8 -*-
\# prova qualcosa come
def _index():
    grid = SQLFORM.grid(
        db.generic.id>0,
        selectable = [('foo', _foo)]
        \# selectable = _foo
    )
    return locals()

def index():
    grid = LOAD('test', '_index.load', ajax=True)
    return locals()
```

2. Make a try with the two values for the selectable parameter (the second one is commented out).

\## What is the expected output? What do you see instead?

using the first value (i.e when the parameter takes a list) and *only using ajax* the callback called _foo is never called.
Everything works perfectly when only the function is passed to the selectable parameter What version of the product are you using? On what operating system? * Framework version: web2py 2.10.0-beta+timestamp.2014.10.16.15.58.50 (Running on Rocket 1.2.6, Python 2.7.5+)
* S.O. version: Linux 3.11-2-amd64 `#1` SMP Debian 3.11.8-1 (2013-11-13) x86_64 GNU/Linux Please provide any additional information below.

_Original issue: http://code.google.com/p/web2py/issues/detail?id=2033_

**Comment by niphlod** [comment]
this is not due to grid's code, but rather on the quite cumbersome fact that jquery submits forms by default via ajax WITHOUT the value of the submit button itself. When you have one button the "value" attribute of the button is not checked in grid's code, while when you have two it's a requirement.
That being said, app/static/js/web2py.js was recently updated to deal with this jquery "discontinuity", in https://github.com/web2py/web2py/pull/540, so in 2.9.12 it's fixed. 

tl;dr:
Just replace the latest web2py.js with the one in your app and you're done
